ChessVariantsLogic/Movement/MoveWorker.cs

```diff
@@ -71,6 +73,8 @@ public GameEvent Move(string move, bool force = false)
 
             if (moves.Contains(coor) || force)
             {
+                var boardTmp = board.CopyBoard();
+                stateLog.Push(boardTmp);
```

Comment:
It is fairly costly to copy the board. If this is only needed for testing, the perhaps we can make this optional somehow? This isn't really a problem for player vs player, but it might be a performance problem for the AI?

I'm open for discussion on this!